File: Data Structures & Algorithms/anagram-groups/submission-3.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:

        mp = {}



        for word in strs:
            res = "".join(sorted(word))

            if(res in mp):
                mp.get(res).append(word)
            else:
                arr = []
                arr.append(word)
                mp[res] = arr

            
        return list(mp.values())
                
            
        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.java
class Solution {
    public int[] productExceptSelf(int[] nums) {


        int[] arr = new int[nums.length];
        for (int i = 0; i<nums.length; i++){
            int s = 1;
            for(int k = 0; k<nums.length; k++){
                if(i != k){
                    s *= nums[k];
                }
            }
           
           arr[i] = s;
        }  

        return arr;
    }
}  
